client/my-sites/checkout/composite-checkout/hooks/use-get-thank-you-url/get-thank-you-page-url.ts

```diff
@@ -125,14 +130,23 @@ export default function getThankYouPageUrl( {
 	const pendingOrReceiptId = getPendingOrReceiptId( receiptId, orderId, purchaseId );
 	debug( 'pendingOrReceiptId is', pendingOrReceiptId );
 
-	// jetpack userless checkout uses a special thank you page
+	// jetpack site-only(userless) checkout needs to first redirect to initiate user conection/authorization
+	// and then to custom thank-you page
 	if ( isJetpackCheckout ) {
-		debug( 'redirecting to userless jetpack thank you' );
+		debug( 'Redirecting to the site to initiate user connection' );
 
 		// extract a product from the cart, in userless checkout there should only be one
 		const productSlug = cart?.products[ 0 ]?.product_slug;
-
-		return `/checkout/jetpack/thank-you/${ siteSlug }/${ productSlug ?? 'no_product' }`;
+		const thankYouPageUrl = `/checkout/jetpack/thank-you/${ siteSlug }/${
+			productSlug ?? 'no_product'
+		}`;
+		const siteAdminUrl = adminUrl || getAdminUrlFromQueryArg();
+		const authorizationUrl = `${ siteAdminUrl }admin.php?page=jetpack&action=authorize_redirect&from=jetpack_site_only_checkout&dest_url=http://wordpress.com${ thankYouPageUrl }`;
+
+		if ( siteAdminUrl ) {
+			return authorizationUrl;
```

Comment:
This looks great to me; could you add a test to cover this new possible return value in https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/composite-checkout/test/composite-checkout-thank-you.js?

Reply:
Oh yes.. forgot about that.  Ok, Done! 👍 🙂